File: core/src/main/java/com/andysklyarov/finnotify/data/CurrencyInRubDataSource.java
package com.andysklyarov.finnotify.data;

import com.andysklyarov.finnotify.domain.CurrencyInRub;

import java.time.LocalDate;

import io.reactivex.Single;

public interface CurrencyInRubDataSource {

    Single<CurrencyInRub> getCurrencyOnDate(String currencyCode, LocalDate date);

    Single<LocalDate> getLastDate();
}

File: core/src/main/java/com/andysklyarov/finnotify/data/CurrencyInRubDataSource.kt
package com.andysklyarov.finnotify.data

import com.andysklyarov.finnotify.domain.CurrencyInRub
import io.reactivex.Single
import java.time.LocalDate

interface CurrencyInRubDataSource {
    fun getCurrencyOnDate(currencyCode: String, date: LocalDate): Single<CurrencyInRub>
    fun getLastDate(): Single<LocalDate>
}